network/server.go

```diff
@@ -590,10 +591,31 @@ func StringToAddrInfo(addr string) (*peer.AddrInfo, error) {
 
 // AddrInfoToString converts an AddrInfo into a string representation that can be dialed from another node
 func AddrInfoToString(addr *peer.AddrInfo) string {
-	if len(addr.Addrs) != 1 {
-		panic("Not supported")
+	// Regex used for matching loopback addresses (IPv4 and IPv6)
+	// This regex will match:
+	// /ip4/localhost/tcp/<port>
+	// /ip4/127.0.0.1/tcp/<port>
+	// /ip4/<any other loopback>/tcp/<port>
+	// /ip6/<any loopback>/tcp/<port>
+	loopbackRegex := regexp.MustCompile(
```

Comment:
You can use same parser every time. You might want to declare as global variable to avoid useless cost.

Reply:
Good call 👍 

Moved it out to a constant:
0bfea14